### DISCUSSION TITLE
Implementing SEB with Moodle SSO for Exam Access

### DISCUSSION THREAD

**Discussion by Doctor-Aiden** [SafeExamBrowser/seb-win-refactoring#950] [Q&A]
Hello!

I am trying to utilize SEB for my university, where we have a Single Sign-On (SSO) login system integrated with Moodle. Our goal is to create a streamlined experience for students where they can launch a SEB config file, log in through SSO, access Moodle, and then proceed directly to their exams.

I’m aware of the template feature in SEB, and I’ve been conducting tests to achieve this setup. My current approach involves creating a template, uploading it to Moodle’s administration, and having instructors select this template when creating an exam. However, the challenge arises because each exam course requires a different config file, making it unfeasible for our intended implementation.

To summarize, here’s what I’m aiming to achieve:

1. **Create a Universal Config File:** This config file would be hosted on our university server and made available for download by all students. When the config is launched, it should open SEB and direct the user to the SSO login page.
2. **Seamless Exam Access:** After logging in, the student would navigate to the exam page in Moodle and start their attempt without any further configurations needed.

I have been struggling to implement this solution for weeks without success. The primary reason I want to set it up this way is to leverage the URL filtering feature in SEB to block certain URLs from the SSO login page all the way through to the exam attempt.

**We are using Moodle 4.1 and SEB config tool 3.5**

Has anyone successfully implemented something similar or could provide guidance on how to achieve this? Any help would be greatly appreciated.

Thank you!

**Comment by dbuechel** [comment]
The following post may provide useful information: #833.

**Reply by Doctor-Aiden** [reply]
> The following post may provide useful information: #833.

Hi Damian

Unfortunately this is not a good approach because it revolves around modifying the Moodle PHP code.

Is there any other way to have a universal config for all exams?

Thank you.

**Reply by dbuechel** [reply]
I unfortunately do not know of any other options, as I myself do not have much experience with Moodle. But you may have more luck asking on a Moodle forum or alike.
